### PR TITLE
Add copy to clipboard to code examples

### PR DESCRIPTION
**Pull request by merelinguist**
_This is my first commit to trpc I hope it’s okay!_ 

Adds a copy button on code examples.

Screenshot 2021-11-29 at 19 42 48

### CODE REVIEW

**Comment by KATT** [comment]
Wow, thank you @merelinguist!! 🙏 🙏

**Comment by KATT** [comment]
Looks great!

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "@trpc/server": "^9.14.0",
     "autoprefixer": "^10.4.0",
     "clsx": "^1.1.1",
+    "copy-to-clipboard": "^3.3.1",
```
Comment (KATT) [inline package.json]:
```suggestion
```

File: src/utils/useClipboard.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+import copy from 'copy-to-clipboard';
```
Comment (KATT) [inline src/utils/useClipboard.ts]:
```suggestion
```

File: src/utils/useClipboard.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      return () => {
+        clearTimeout(id);
+      };
+    }
+  }, [hasCopied]);
+
+  return [
+    hasCopied,
+    () => {
+      setHasCopied(copy(text));
```
Comment (KATT) [inline src/utils/useClipboard.ts]:
```suggestion
      setHasCopied(true);
      navigator.clipboard.writeText(text);
```

**Review by KATT — CHANGES_REQUESTED** [review]
Major browser seem to have support for the clipboard API, could we possibly use that?

https://caniuse.com/?search=clipboard

**Comment by KATT** [comment]
You're [added as a contributor](https://github.com/trpc/examples-kitchen-sink/invitations) so if you feel like doing more stuff, you can make branches easier :)

**Comment by merelinguist** [comment]
> Major browser seem to have support for the clipboard API, could we possibly use that?
> 
> https://caniuse.com/?search=clipboard

Done :)

**Review by KATT — APPROVED** [review]

**Pull request merged**